packages/core/saved-objects/core-saved-objects-server/src/saved_objects_type.ts

```diff
@@ -56,14 +57,24 @@ export interface SavedObjectsType<Attributes = any> {
   mappings: SavedObjectsTypeMappingDefinition;
   /**
    * An optional map of {@link SavedObjectMigrationFn | migrations} or a function returning a map of {@link SavedObjectMigrationFn | migrations} to be used to migrate the type.
+   * @deprecated Will throw if registering migrations for version higher than 8.??.00
+   *             Use {@link SavedObjectsModelVersion | modelVersions} instead
    */
   migrations?: SavedObjectMigrationMap | (() => SavedObjectMigrationMap);
+  /**
+   * An optional map of {@link SavedObjectsModelVersion}
+   *
+   * @remarks After version 8.??.00, migrations are deprecated and using model versions instead
+   *          will be the only way to define mutation of data/schema for saved object types.
+   */
+  modelVersions?: SavedObjectsModelVersionMap | SavedObjectsModelVersionMapFn;
```

Comment:
Given the change of structure, we will switch to using a new property for the concept of 'model version'.

Which means that we will throw if a consumer tries to register a migration for a version higher than the milestone / transition version to model versioning.